File: src/main/java/com/lalkalol/mujahid/commands/impl/RemoveCommand.java
package com.lalkalol.mujahid.commands.impl;

import com.lalkalol.mujahid.commands.Command;
import com.lalkalol.mujahid.commands.CommandContext;
import dev.arbjerg.lavalink.client.player.Track;
import net.dv8tion.jda.api.interactions.commands.OptionType;
import net.dv8tion.jda.api.interactions.commands.build.Commands;
import net.dv8tion.jda.api.interactions.commands.build.OptionData;
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData;

public class RemoveCommand implements Command {
    @Override
    public String name() {
        return "remove";
    }

    @Override
    public SlashCommandData data() {
        return Commands.slash("remove", "Remove a track from the queue by position.")
                .addOptions(
                        new OptionData(OptionType.INTEGER, "position", "Queue position (see /queue)", true)
                                .setMinValue(1)
                );
    }

    @Override
    public void execute(CommandContext ctx) {
        int position = ctx.getEvent().getOption("position").getAsInt();
        Track removed = ctx.getMusicManager().getScheduler().removeAt(position - 1);
        if (removed == null) {
            ctx.replyEphemeral("There is no track at position " + position + ".");
        } else {
            ctx.reply("Removed **" + (removed.getInfo().getTitle().isBlank() ? "Unknown" : removed.getInfo().getTitle())
                    + "** from the queue.");
        }
    }
}

File: src/main/kotlin/com/lalkalol/mujahid/commands/impl/RemoveCommand.kt
package com.lalkalol.mujahid.commands.impl

import com.lalkalol.mujahid.commands.Command
import com.lalkalol.mujahid.commands.CommandContext
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.Commands
import net.dv8tion.jda.api.interactions.commands.build.OptionData
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData

class RemoveCommand : Command {
    override fun name(): String = "remove"

    override fun data(): SlashCommandData =
        Commands.slash("remove", "Remove a track from the queue by position.")
            .addOptions(
                OptionData(OptionType.INTEGER, "position", "Queue position (see /queue)", true)
                    .setMinValue(1),
            )

    override suspend fun execute(ctx: CommandContext) {
        val position = ctx.event.getOption("position")!!.asInt
        val removed = ctx.getMusicManager().scheduler.removeAt(position - 1)
        if (removed == null) {
            ctx.replyEphemeral("There is no track at position $position.")
        } else {
            val title = if (removed.info.title.isBlank()) "Unknown" else removed.info.title
            ctx.reply("Removed **$title** from the queue.")
        }
    }
}
